src/cache/cache_las.c

```diff
@@ -785,46 +705,107 @@ __wt_las_insert_updates(WT_CURSOR *cursor, WT_BTREE *btree, WT_PAGE *page, WT_MU
+                if (modifies.size > 0) {
+                    __wt_modify_vector_pop(&modifies, &upd);
+                    WT_ASSERT(session, upd->type == WT_UPDATE_STANDARD);
+                } else
+                    continue;
+            }
+            full_value->data = upd->data;
+            full_value->size = upd->size;
+        }
+
+        /* Flush the updates on stack */
+        for (; modifies.size > 0; tmp = full_value, full_value = prev_full_value,
+                                  prev_full_value = tmp, upd = prev_upd) {
+            /* Should not see BIRTHMARK or TOMBSTONE */
+            WT_ASSERT(session, upd->type == WT_UPDATE_STANDARD || upd->type == WT_UPDATE_MODIFY);
+
+            __wt_modify_vector_pop(&modifies, &prev_upd);
+            stop_ts_pair.timestamp = prev_upd->start_ts;
+            stop_ts_pair.txnid = prev_upd->txnid;
+
+            /* Skip the updates have the same start timestamp and transaction id */
+            if (upd->start_ts != prev_upd->start_ts || upd->txnid != prev_upd->txnid) {
+                if (!insert_modify)
+                    WT_ERR(__las_insert_update(session, cursor, btree_id, key, upd,
+                      WT_UPDATE_STANDARD, full_value, stop_ts_pair));
+                else if (__las_insert_update(session, cursor, btree_id, key, upd, WT_UPDATE_MODIFY,
+                           modify_value, stop_ts_pair) != 0) {
+                    __wt_scr_free(session, &modify_value);
+                    goto err;
+                } else
+                    __wt_scr_free(session, &modify_value);
+
+                ++insert_cnt;
+            } else
+                squashed = true;
+
+            if (prev_upd->type == WT_UPDATE_TOMBSTONE) {
+                WT_ASSERT(session, modifies.size > 0);
+                __wt_modify_vector_pop(&modifies, &prev_upd);
+
+                /* The update newer to a TOMBSTONE must be a full update */
+                WT_ASSERT(session, prev_upd->type == WT_UPDATE_STANDARD);
+            }
+
+            if (prev_upd->type == WT_UPDATE_MODIFY) {
+                WT_ERR(__wt_buf_set(session, prev_full_value, full_value->data, full_value->size));
+                WT_ERR(__wt_modify_apply_item(session, prev_full_value, prev_upd->data, false));
+            } else {
+                WT_ASSERT(session, prev_upd->type == WT_UPDATE_STANDARD);
 
-                /* Do not put birthmarks into the lookaside. */
+                prev_full_value->data = prev_upd->data;
+                prev_full_value->size = prev_upd->size;
+                insert_modify = false;
                 continue;
             }
 
-            if (upd->prepare_state == WT_PREPARE_INPROGRESS)
-                ++prepared_insert_cnt;
+            /* Calculate reverse delta */
+            nentries = MAX_REVERSE_MODIFY_NUM;
+            if (__wt_calc_modify((WT_SESSION *)session, prev_full_value, full_value,
+                  prev_full_value->size / 10, entries, &nentries) != 0)
```

Comment:
Here I follow mongodb's usage, which limits the delta size to be the 1/10 of the base value. We probably should make this configurable in the future.